extends CharacterBody2D


var chase = false
var speed = 100
var alive = true

@onready var animation = $AnimatedSprite2D

func _physics_process(delta: float) -> void:
	# Add the gravity.
	if not is_on_floor():
		velocity += get_gravity() * delta
		
	var player = $"../../Player/Player"
	var direction = (player.position - self.position).normalized()
	
	if alive:
		if chase: 
			velocity.x = direction.x * speed
			animation.play("Run")
		else:
			velocity.x = 0;
			animation.play("Idle")


		if direction.x < 0:
			animation.flip_h = true
		else:
			animation.flip_h = false

	move_and_slide()


func _on_detector_body_entered(body: Node2D) -> void:
	if body.name == 'Player':
		chase = true


func _on_detector_body_exited(body: Node2D) -> void:
	if body.name == 'Player':
		chase = false


func _on_death_body_entered(body: Node2D) -> void:
	if body.name == 'Player':
		body.velocity.y -= 200
		death()
		
func _on_death_2_body_entered(body: Node2D) -> void:
	if body.name == 'Player':
		if alive:
			body.health -= 40
		death()
	
func death():
	alive = false
	animation.play("Death")
	await animation.animation_finished
	queue_free()
